protocols/cosi/cosi.go

```diff
@@ -234,18 +221,8 @@ func (pc *ProtocolCosi) handleCommitment(in *CosiCommitment) error {
 	return pc.SendTo(pc.Parent(), outMsg)
 }
 
-type ChallengeHook func(*CosiChallenge) error
-
 // StartChallenge start the challenge phase. Typically called by the Root ;)
 func (pc *ProtocolCosi) StartChallenge() error {
-	// first check the hook
-	/*if pc.challengeHook != nil {*/
-	//return pc.challengeHook(nil)
-	/*}*/
-
-	if pc.Message == nil {
-		return fmt.Errorf("%s StartChallenge() called without message (=%v)", pc.Node.Name(), pc.Message)
-	}
```

Comment:
The reason it was put here is because there can be two modes: 
* either put the message in the Announcement
* either put the message in the Challenge. 
The second approach has the benefit of being able to correlate rounds like BizCoin where the Response of the first round is sent as the message in the Challenge of the second round. The first approach lets users make the verification as soon as possible.
So Both are correct and only the latter is used here.
Regarding the empty message... I thought it would be considered invalid but in fact, it seems gpg and others don't complain with an empty message, so why should we ... ! OK